### ISSUE TITLE
API - CLI

### ISSUE THREAD

**Issue by bebraw** [webpack/webpack.io#24]
[Stub](https://github.com/webpack/webpack.io/blob/develop/content/api/cli.md).

Feel free to comment here if you have ideas on what this guide should cover. Link to potential resources too.

**Comment by pksjce** [comment]
Can I take this one up?
Played around with the cli and have found a few limitations and one bug in its usage.
1. config, entry, output options worked pretty well.
2. I was able to add babel loader through --module-bind. But this was limited because we cannot add any other params than test and loader name.(esp excludes)
3. I tried out the --provide option. This has a bug because 
`options.plugins.push(new ProvidePlugin(name, value));` is wrong. ProvidePlugin expects a key value pair.
4. There are no test cases for the cli.
5. The --help of the cli kinda sucks, even though webpack uses yargs which is extremely flexible and provides rich features.

What is the future for the cli? Will it be deprecated completely? Or is it in the roadmap for improvement?

Personally, I feel its a great tool for quick trials on your build before transferring permanently to a config file. Also, if there is support for a feature, there must be full support, not half implemented or broken, right?

Bottom line is I'm not sure how much documentation is merited to this section?

**Comment by bebraw** [comment]
@pksjce Sure. The [current documentation](https://webpack.github.io/docs/cli.html) would be a good starting point. I agree the CLI is difficult and poorly documented at the moment and I see a lot of problems around it as people don't understand what kind of configuration it generates. Often I end up checking the source to find the answers.

@sokra @TheLarkInn Any thoughts on this? How do you see the future of the CLI?

**Comment by pksjce** [comment]
@bebraw - Yes, I was exploring each of the options mentioned in the current documentation to see if I could add anything useful to the info already present.
Meanwhile, progress, profile, stats and watch flags fall into the advantages of using the cli. They are great features and I really learnt a lot about my codebase exploring these flags.
I'll wait for more info on the direction for this.

**Comment by pksjce** [comment]
*Bump*

**Comment by TheLarkInn** [comment]
I agree with @bebraw, start by taking the existing documentation, and then replace any of your findings to update the current functionality.
